tests/unit/train/test_eagle3_attention.py

```diff
@@ -0,0 +1,139 @@
+# ruff: noqa: ERA001
+import pytest
```

Comment:
Do we have tests for all the other training, checkpointing, etc utilities that you've written?